hudi-common/src/main/java/org/apache/hudi/common/fs/inline/InLineFsDataInputStream.java

```diff
@@ -56,24 +60,32 @@ public long getPos() throws IOException {
 
   @Override
   public int read(long position, byte[] buffer, int offset, int length) throws IOException {
+    if ((length - offset) > this.length) {
```

Comment:
I think the condition be length + offset > this.length. Right ? Can you please check

Reply:
@nsivabalan : ^^